CONTRIBUTING.md

```diff
@@ -8,11 +8,10 @@ Please see https://git.k8s.io/community/CLA.md for more info.
 
 ## Prerequisites
 - [go](https://golang.org/dl/) version v1.13+.
-- [dep](https://github.com/golang/dep) dep v0.5+
 - [docker](https://docs.docker.com/install/) version 17.03+.
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.11.3+.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.14.1+.
 - [kustomize](https://sigs.k8s.io/kustomize/docs/INSTALL.md) v3.1.0+
-- Access to a Kubernetes v1.11.3+ cluster.
+- Access to a Kubernetes v1.14.1+ cluster.
```

Comment:
Do we need 1.14.1 for v2? Which feature is required for v2 that was added by 1.14.1?

Reply:
We have no tests to ensure the usage of the project with old versions < 1.14.1+.

Also, I know that from v1.11.+ to the latest has significant changes, however, it is very hard to know what feature could or not been impacted by it.  And then, IMO make sense we keep the lower version which we are using in the CI to verify the supportability of it.

Reply:
But the hard requirement set in code is 1.11, what about saying 1.11.3+ is needed, 1.14.1+ is recommended as that is the lowest one with automatic tests?

Reply:
It is true. I will revert since it is not part of the scope of this PR as well.

Reply:
Done.